### PR TITLE
block_discard: Add a step for checking partitions

### PR DESCRIPTION
**Pull request by YongxueHong**
Add a step to get partitions in during time continuously since
it will spend some time in syncing the operation of
creating partitions.

ID: 1695500
Signed-off-by: Yongxue Hong <yhong@redhat.com>

### CODE REVIEW

**Comment by YongxueHong** [comment]
@hereischen  @CongLi  @XueqiangWei  @ngu-niny pls help me review.

File: qemu/tests/block_discard.py [hunk 1]
```diff
@@ -114,11 +121,15 @@ def get_allocation_bitmap():
...
-    format_disk_cmd = params["format_disk_cmd"]
-    format_disk_cmd = format_disk_cmd.replace("DISK", device_name)
-    error_context.context("format disk '%s' in guest" % device_name,
+    error_context.context("Create partition on '%s' in guest" % device_name,
                           logging.info)
-    session.cmd(format_disk_cmd)
+    session.cmd(params['create_partition_cmd'].replace("DISK", device_name))
+
+    if not utils_misc.wait_for(_check_disk_partitions_number, 30, step=3.0):
+        test.error('Failed to get 1 partitions on %s.' % device_name)
```
Comment (hereischen) [inline qemu/tests/block_discard.py]: ` 1 partition`

File: qemu/tests/cfg/block_discard.cfg [hunk 2]
```diff
@@ -22,11 +22,11 @@
...
+            create_partition_cmd = "parted -s DISK mklabel msdos && "
+            create_partition_cmd += "parted -s DISK mkpart primary ext2 2048s 100%"
+            format_disk_cmd = "mkfs.xfs -f DISK1"
         - ext4:
             no RHEL.3 RHEL.4 RHEL.5
-            format_disk_cmd = "parted DISK mklabel msdos &&"
-            format_disk_cmd += "parted DISK mkpart primary ext2 2048s 100% &&"
-            format_disk_cmd += "yes|mkfs.ext4 DISK1"
+            create_partition_cmd = "parted -s DISK mklabel msdos && "
+            create_partition_cmd += "parted -s DISK mkpart primary ext2 2048s 100%"
```
Comment (CongLi) [inline qemu/tests/cfg/block_discard.cfg]: ext2 -> ext4

File: qemu/tests/cfg/block_discard.cfg [hunk 3]
```diff
@@ -22,11 +22,11 @@
     variants:
         - xfs:
             no RHEL.3 RHEL.4 RHEL.5 RHEL.6
-            format_disk_cmd = "parted DISK mklabel msdos &&"
-            format_disk_cmd += "parted DISK mkpart primary ext2 2048s 100% &&"
-            format_disk_cmd += "mkfs.xfs -f DISK1"
+            create_partition_cmd = "parted -s DISK mklabel msdos && "
+            create_partition_cmd += "parted -s DISK mkpart primary ext2 2048s 100%"
```
Comment (CongLi) [inline qemu/tests/cfg/block_discard.cfg]: ext2 -> xfs

File: qemu/tests/block_discard.py (hunk 1 above)
Comment (CongLi) [reply qemu/tests/block_discard.py]: failed to get partition is fine, '1' is a little wired here.

File: qemu/tests/cfg/block_discard.cfg (hunk 2 above)
Comment (YongxueHong) [reply qemu/tests/cfg/block_discard.cfg]: thanks, have updated.

File: qemu/tests/cfg/block_discard.cfg (hunk 3 above)
Comment (YongxueHong) [reply qemu/tests/cfg/block_discard.cfg]: thanks, have updated.

File: qemu/tests/block_discard.py (hunk 1 above)
Comment (YongxueHong) [reply qemu/tests/block_discard.py]: thanks, have updated.

**Comment by XueqiangWei** [comment]
LGTM,
Acked-by xuwei@redhat.com

**Comment by ngu-niny** [comment]
Test PASS on ppc:

'01-Host_RHEL.m8.u0.qcow2.spapr_vscsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.block_discard.xfs.unmap' Finshed 2019-04-11 03:50:48 2019-04-11 03:52:15 PASS 87.0845
'02-Host_RHEL.m8.u0.qcow2.spapr_vscsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.block_discard.xfs.writesame' Finshed 2019-04-11 03:52:16 2019-04-11 03:53:44 PASS 88.40591
'03-Host_RHEL.m8.u0.qcow2.spapr_vscsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.block_discard.ext4.unmap' Finshed 2019-04-11 03:53:44 2019-04-11 03:55:09 PASS 84.868394
'04-Host_RHEL.m8.u0.qcow2.spapr_vscsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.block_discard.ext4.writesame' Finshed 2019-04-11 03:55:09 2019-04-11 03:56:33 PASS 84.228036

Acked-by: Nini Gu <ngu@redhat.com>

**Comment by CongLi** [comment]
Ack.
Finshed=4000, PASS=4000

**Comment by YongxueHong** [comment]
thanks all

**Pull request merged**